src/gui/guiButtonImage.cpp

```diff
@@ -32,15 +32,15 @@ using namespace gui;
 GUIButtonImage::GUIButtonImage(gui::IGUIEnvironment *environment,
 		gui::IGUIElement *parent, s32 id, core::rect<s32> rectangle,
 		ISimpleTextureSource *tsrc, bool noclip)
-	: GUIButton (environment, parent, id, rectangle, tsrc, noclip)
+	: GUIButton(environment, parent, id, rectangle, tsrc, noclip)
 {
-	m_image = Environment->addImage(
-			core::rect<s32>(0,0,rectangle.getWidth(),rectangle.getHeight()), this);
-	m_image->setScaleImage(isScalingImage());
+	GUIButton::setScaleImage(true);
+	m_image = new GUIAnimatedImage(environment, this, id, rectangle);
```

Comment:
irr_ptr?

Reply:
Not possible without changing a lot more code like `m_clickthrough_elements`, better leave it to dedicated PRs.